integrations/db/src/utils/sync-game-state.ts

```diff
@@ -194,33 +177,165 @@ export function syncGameState(
         state.items.delete(item.id);
       }
     }
-    logger.debug({ characterId }, "Character left game service via db poll");
+    logger.debug(
+      { characterId },
+      "Character left game service via Electric sync",
+    );
   }
-}
 
-async function getOnlineCharacterIdsForAreaFromDb(
-  drizzle: DrizzleClient,
-  areaId: AreaId,
-): Promise<ReadonlySet<CharacterId>> {
-  return new Set(
-    (
-      await drizzle
-        .select({ id: characterTable.id })
+  /**
+   * Handle character changes from Electric
+   */
+  async function handleCharacterChanges(
+    charactersData: Map<string, Record<string, unknown>>,
+  ) {
+    const dbIds = new Set(charactersData.keys());
+    const stateIds = characterIdsInState(state);
+    const removedIds = stateIds.difference(dbIds);
+    const addedIds = dbIds.difference(stateIds);
+
+    // Remove characters that are no longer in the DB
+    removedIds.forEach(removeCharacterFromGameState);
+
+    // Add new characters from the DB
+    if (addedIds.size > 0) {
+      const addedCharacters = await drizzle
```

Comment:
why are you using both drizzle and eletric, you should be able to use only electric